src/main/java/com/synopsys/integration/detect/tool/detector/DetectorToolResult.java

```diff
@@ -37,13 +37,57 @@
 import com.synopsys.integration.util.NameVersion;
 
 public class DetectorToolResult {
-    public Optional<NameVersion> bomToolProjectNameVersion = Optional.empty();
-    public List<DetectCodeLocation> bomToolCodeLocations = new ArrayList<>();
+    private final Optional<NameVersion> bomToolProjectNameVersion;
```

Comment:
Optional should not be a field. Make it a regular NameVersion. Annotate the field with `@Nullable` as well as the constructor paramter. Use `Optional.ofNullable(bomToolProjectNameVersion)` in `getBomToolProjectNameVersion()`.

Do the same for `rootDetectorEvaluationTree` below.